jax/api.py

```diff
@@ -443,6 +466,31 @@ def fun(*args):
   return apply_jaxtree_fun(fun, io_tree, *py_args)
 
 def vjp(fun, *primals):
+  """Compute a (reverse-mode) vector-Jacobian product of `fun`.
+
+  This is a more general form of `grad` that can be used for functions with
+  non-scalar outputs. For most common use cases, you most likely want `grad`
+  instead of `vjp`.
+
+  Args:
+    fun: Function to be differentiated. Its arguments should be arrays, scalars,
+      or standard Python containers of arrays or scalars. It should return an
+      array, scalar, or standard Python container of arrays or scalars.
+    primals: A sequence of of primal values at which the Jacobian of `fun`
+      should be evaluated. Should be a tuple of arrays, scalar, or standard
+      Python containers thereof.
+
+  Returns:
+    A `(primals_out, gradient)` pair, where `primals_out` is `fun(*primals)`.
+    `gradient` is a function from tangent values that computes vector-Jacobian
+    product of `fun` in an epsilon ball around `primals`.
```

Comment:
Could add that the gradient fun (maybe call it vjp_at_primals?) accepts an argument cotangent vector with the same shape as primals_out. 

Instead if “in an epsilon ball around” I think we should just write “at”. Could also say it returns a cotangent vector with the same shape as primals.

Reply:
Maybe instead of "tangent values" say "cotangent values". We readily identify all these things but that language can be a kind of type hint to the careful reader.

Reply:
Taking those suggestions together, we might end up with:

A `(primals_out, vjpfun)` pair, where `primals_out` is `fun(*primals)`. `vjpfun` is a function from a cotangent vector with the same shape as `primals_out` to a tuple of cotangent vectors with the same shape as `primals`, representing the vector-Jacobian product of `fun` evaluated at `primals`.